pages/_app.js

```diff
@@ -1,15 +1,41 @@
 import '../styles/globals.css';
+import React from 'react';
 import { CookiesProvider } from 'react-cookie';
 import { ChakraProvider } from '@chakra-ui/react';
+import { Provider } from 'next-auth/client';
+import { useSession, signIn } from 'next-auth/client';
 
 function MyApp({ Component, pageProps }) {
   return (
     <ChakraProvider>
       <CookiesProvider>
-        <Component {...pageProps} />
+        <Provider session={pageProps.session}>
+          {Component.auth ? (
+            <Auth>
+              <Component {...pageProps} />
+            </Auth>
+          ) : (
+            <Component {...pageProps} />
+          )}
+        </Provider>
       </CookiesProvider>
     </ChakraProvider>
   );
 }
 
+function Auth({ children }) {
+  const [session, loading] = useSession();
+  const isUser = !!session?.user;
+  React.useEffect(() => {
```

Comment:
Any reason to not import `useEffect` above and have this scoped as `React.useEffect`?

Reply:
no, will change it :)